lib/ts/recipe/webauthn/types.ts

```diff
@@ -0,0 +1,608 @@
+            rawId: string;
+            response: {
+                clientDataJSON: string;
+                attestationObject: string;
+                transports?: ("ble" | "cable" | "hybrid" | "internal" | "nfc" | "smart-card" | "usb")[];
+                userHandle: string;
+            };
+            authenticatorAttachment: "platform" | "cross-platform";
+            clientExtensionResults: Record<string, unknown>;
+            type: "public-key";
+        };
+        session: SessionContainerInterface | undefined;
+        shouldTryLinkingWithSessionUser: boolean | undefined;
+        tenantId: string;
+        userContext: UserContext;
+    }): Promise<
+        | { status: "OK"; user: User; recipeUserId: RecipeUserId }
+        | { status: "WRONG_CREDENTIALS_ERROR" }
+        // when the attestation is checked and is not valid or other cases in which the authenticator is not correct
+        | { status: "INVALID_AUTHENTICATOR_ERROR" }
+        | {
+              status: "LINKING_TO_SESSION_USER_FAILED";
+              reason:
+                  | "EMAIL_VERIFICATION_REQUIRED"
+                  | "RECIPE_USER_ID_ALREADY_LINKED_WITH_ANOTHER_PRIMARY_USER_ID_ERROR"
+                  | "ACCOUNT_INFO_ALREADY_ASSOCIATED_WITH_ANOTHER_PRIMARY_USER_ID_ERROR"
+                  | "SESSION_USER_ACCOUNT_INFO_ALREADY_ASSOCIATED_WITH_ANOTHER_PRIMARY_USER_ID_ERROR";
+          }
+    >;
+
+    /**
+     * We pass in the email as well to this function cause the input userId
+     * may not be associated with an webauthn account. In this case, we
+     * need to know which email to use to create an webauthn account later on.
+     */
+    generateRecoverAccountToken(input: {
+        userId: string; // the id can be either recipeUserId or primaryUserId
+        email: string;
+        tenantId: string;
+        userContext: UserContext;
+    }): Promise<
+        { status: "OK"; token: string } | { status: "UNKNOWN_USER_ID_ERROR" } | { status: "UNKNOWN_EMAIL_ERROR" }
+    >;
+
+    // make sure the email maps to options email
+    consumeRecoverAccountToken(input: {
+        token: string;
+        webauthnGeneratedOptionsId: string;
+        credential: {
+            id: string;
+            rawId: string;
+            response: {
+                clientDataJSON: string;
+                attestationObject: string;
+                transports?: ("ble" | "cable" | "hybrid" | "internal" | "nfc" | "smart-card" | "usb")[];
+                userHandle: string;
+            };
+            authenticatorAttachment: "platform" | "cross-platform";
+            clientExtensionResults: Record<string, unknown>;
+            type: "public-key";
+        };
+        tenantId: string;
+        userContext: UserContext;
+    }): Promise<
+        | {
+              status: "OK";
+              email: string;
+              userId: string;
+          }
+        | { status: "WRONG_CREDENTIALS_ERROR" }
+        | { status: "RECOVER_ACCOUNT_TOKEN_INVALID_ERROR" }
```

Comment:
shouldnt this also have `INVALID_AUTHENTICATOR_ERROR` as an output?

Reply:
done